src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -417,7 +413,7 @@ private HaxeClassResolveResult resolveHaxeClassInternal() {
       }
 
       return resolveResult;
-    }
+    } // end (this instanceof HaxeArrayLiteral)
```

Comment:
I think we should divide code into smaller portions in the future, to avoid such comments

Reply:
These kinds of comments bother you?  😮  That's interesting.

While I agree that we should have smaller routines, in general, I don't want to make any hard rules about this.  Sometimes it is easier to see an algorithm when it is all kept in one routine; sometimes it's not.  At any rate, I am not a person who hates comments.  Their only downfall is that people who maintain the code don't later fix them when things change.

Not that you mention that, though, I'm thinking about adding a decorating feature (probably another plugin, since I need it in Java) that auto-adds this type of comment.  Not that it would add it to the source, but that it would add it when on my screen.  Of course, that does nothing to help code reviews in github, which is where I actually need it most of the time.

Reply:
In my opinion, the situation when need for such kind of comments appears, suggests that the method is too complex and performs too much work - meaning of its actions becomes ambiguous.

Separation of such methods to less complex is considered good practice. It's rises code readability and the possibility of re-using.